File: go/clock/clock.go
package clock

import "fmt"

const TestVersion = 2
const wholeDay = 24 * 60

type Clock struct {
	hour   int
	minute int
}

func Time(hour, minute int) Clock {
	var midnight = Clock{0, 0}
	var allMinutes = (hour * 60) + minute
	return midnight.Add(allMinutes)
}

func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.hour, c.minute)
}

func (c Clock) Add(minute int) Clock {
	var minutes = (c.minutes() + minute) % wholeDay
	if minutes < 0 {
		minutes = wholeDay + minutes
	}
	var h, m = divmod(minutes, 60)
	return Clock{h % 24, m}
}

func (c Clock) minutes() int {
	return c.hour*60 + c.minute
}

func divmod(x, y int) (int, int) {
	return x / y, x % y
}

File: python/clock/clock.py
# The clock wraps from 23:59 to 00:00
MAX_MINUTES = (24 * 60)
class Clock:
    def __init__(self, hour, minute):
        self.raw_minutes = (hour * 60) + minute

    def __str__(self):
        h = self.minutes // 60
        m = self.minutes % 60
        return "{:02}:{:02}".format(h, m)

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.minutes == other.minutes
        else:
            False

    def add(self, minutes):
        return Clock(0, self.minutes + minutes)

    @property
    def minutes(self):
        """
        Number of minutes on the clock, wrapped to number of minutes in the
        day.
        """
        return self.raw_minutes % MAX_MINUTES

